src/java/org/apache/cassandra/db/SinglePartitionReadCommand.java

```diff
@@ -1266,6 +1272,32 @@ public boolean isRangeRequest()
         return false;
     }
 
+    /*
+     * The execution method does not need to perform reconciliation so the read command
+     * should execute in a mannager suited to not needing reconciliation. Such as when
+     * executing transactionally at a single replica and doing an index scan where the index
+     * scan should not return extra rows and expect post filtering at the coordinator.
+     */
+    public SinglePartitionReadCommand withoutReconciliation()
+    {
+        if (indexQueryPlan() == null)
+            return this;
```

Comment:
When [CASSANDRA-19007](https://issues.apache.org/jira/browse/CASSANDRA-19007) gets done, I think we could have a filtering queries that will be controlled by the row filter's reconciliation flag as well. We may want to simply check `rowFilter().isEmpty()` here. Normal partition and clustering restrictions won't be in the row filter, IIRC

Reply:
Not 100% following, so there are more cases where we want to not have reconciliation, and the right check is `rowFilter().isEmpty()`?

Reply:
Yeah, if you had a partition-restricted non-index filtering query, the row filter would be non-empty and you'd want to get past this short-circuit. Looks like you've already updated this in your last commit though.